### DISCUSSION TITLE
ESP32-WROOM End of Life?

### DISCUSSION THREAD

**Discussion by clarkbremer** [pschatzmann/ESP32-A2DP#231] [Q&A] [answered]
I saw on Mouser today that the ESP32-WROOM-32D module is now "Not recommended for new designs."  Does anyone have any more info, or know what module would be the replacement?  I'm in the process of cost-reducing my prototype, and was planning to drop the module directly on the PCB without using a dev board.  CB.

**Comment by pschatzmann** [comment, marked as answer]
I am not sure, would this help: https://products.espressif.com/#/product-selector?language=en&names=

**Comment by clarkbremer** [comment]
Perfect, thanks.  Seems like it will be `ESP32-WROOM-32E-N4`, which appears to be pin and feature compatible, and not yet available (at least from Mouser).  So looks like it's safe to design around the 32D for now.

**Comment by smagtech** [comment]
I'm getting this error can anybody help please. 

Arduino
Version: 2.2.1
CLI Version: 0.34.0

error is....

first red flag....

In file included from c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioLibs/AudioKit.h:4,
                 from C:\Users\G_Sudhir\AppData\Local\Temp\.arduinoIDE-unsaved20231012-3424-11tccjn.j7oj\basic-a2dp-audiokit\basic-a2dp-audiokit.ino:10:
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h: In member function 'i2s_config_t audiokit::AudioKitConfig::i2sConfig()':
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:123:9: warning: missing initializer for member 'i2s_driver_config_t::fixed_mclk' [-Wmissing-field-initializers]
         };
         ^
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:123:9: warning: missing initializer for member 'i2s_driver_config_t::mclk_multiple' [-Wmissing-field-initializers]
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:123:9: warning: missing initializer for member 'i2s_driver_config_t::bits_per_chan' [-Wmissing-field-initializers]


second red flag.... 


c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp: In member function 'virtual void BluetoothA2DPSinkQueued::i2s_task_handler(void*)':
c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp:52:12: error: unused variable 'bytes_written' [-Werror=unused-variable]
     size_t bytes_written = 0;
            ^~~~~~~~~~~~~
cc1plus.exe: some warnings being treated as errors
c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSink.cpp: In member function 'virtual void BluetoothA2DPSink::audio_data_callback(const uint8_t*, uint32_t)':
c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSink.cpp:1150:10: error: variable 'is_callback_used' set but not used [-Werror=unused-but-set-variable]
     bool is_callback_used = false;
          ^~~~~~~~~~~~~~~~
cc1plus.exe: some warnings being treated as errors



exit status 1

Compilation error: exit status 1
